### PR TITLE
Tyler iteration3

### PR DESCRIPTION
**Pull request by reid-andrew**
Merging Tyler's Iteration 3 work into dev branch.

I'm going to open this PR, but we should review collectively before merging since GH is indicating it can't automatically merge (as expected).

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -70,7 +72,7 @@ def average_goals_per_game
 
   def total_goals_per_season
     Game.all.reduce({}) do |season_goals, game|
-      if !season_goals[game.season.to_s]
+      if season_goals[game.season.to_s].nil?
```
Comment (Joshua-Tukman) [inline lib/stat_tracker.rb:79]: Using the .nil? method makes this more readable

File: test/fixtures/game_teams_smaller_sample.csv [hunk 2]
```diff
@@ -0,0 +1,11 @@
+game_id,team_id,HoA,result,settled_in,head_coach,goals,shots,tackles,pim,powerPlayOpportunities,powerPlayGoals,faceOffWinPercentage,giveaways,takeaways
```
Comment (Joshua-Tukman) [inline test/fixtures/game_teams_smaller_sample.csv:1]: Good call on making this smaller sample files!

**Review by Joshua-Tukman — COMMENTED** [review]
These smaller sample csv files are awesome!

**Review by tylerpporter — APPROVED** [review]
Resolving merge conflicts should just be a matter of moving the new methods from tyler_iteration3 beneath our existing code. For all duplicate methods, default to the existing on dev branch.

**Pull request merged**